apps/hash-frontend/src/shared/table-header.tsx

```diff
@@ -350,6 +334,9 @@ export const TableHeader: FunctionComponent<TableHeaderProps> = ({
             </Box>
           </Box>
         </Box>
+        {generateCsvFile ? (
+          <ExportToCsvButton generateCsvFile={generateCsvFile} />
```

Comment:
Is there a simple default export function we can use that just exports exactly what is rendered in the table? e.g. for use in the type tables

Reply:
Not at the moment, but it shouldn't be too hard to extract a base `generateCsvFromTable` method which is used in the `TableHeader` by default, and then via props specify a second `getAdditionalCsvData` method that fetches the additional data for the entities table. Should I add this as part of this PR?

Reply:
I think we might as well, since it's a cheap way of getting an export option for any other tables also

Reply:
You could also just default to `generateCsvFromTable` if a custom function isn't passed (if `getAdditionalCsvData` becomes a pain)

Reply:
I've added the export to CSV button to the `TableHeader` component (and therefore the types tables) in https://github.com/hashintel/hash/pull/3940/commits/ec3e8dee8200dea9e89699d11408586030bb4193